python/dlr/api.py

```diff
@@ -342,3 +343,37 @@ def get_input(self, name, shape=None):
                                      out.ctypes.data_as(ctypes.POINTER(ctypes.c_float))))
         out = out.reshape(shape)
         return out
+
+# Interface
+class IDLRModel:
+    __metaclass__=abc.ABCMeta
+
+    @abc.abstractmethod
+    def get_input_names(self):
+        return
```

Comment:
throw https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
+1
do it for the other abstract methods that shouldn't be reached as well